post116-business-directory/includes/class-rest.php

```
@@ -0,0 +1,396 @@
+    }
+
+    // debug logging removed
+
+    public static function search(WP_REST_Request $req) {
+        $q = sanitize_text_field($req->get_param('q'));
+        $category = sanitize_text_field($req->get_param('category'));
+        $flags = (array)$req->get_param('flags');
+        $per_page = max(1, min(50, (int)$req->get_param('per_page')));
+        $page = max(1, (int)$req->get_param('page'));
+        global $wpdb;
+        $like_q    = $q !== '' ? '%' . $wpdb->esc_like($q) . '%' : '';
```

PHPMD (2.15.0)

Avoid unused local variables such as '$like_q'. (Unused Code Rules)

(UnusedLocalVariable)